src/authorizations/utils/permissions.ts

```diff
@@ -135,3 +135,72 @@ export enum BucketTab {
   AllBuckets = 'All Buckets',
   Scoped = 'Scoped',
 }
+
+export const formatPermissionsObj = permissions => {
+  const newPerms = permissions.reduce((acc, {action, resource}) => {
+    const {type, id, orgID, name} = resource
+    let p
+
+    if (acc.hasOwnProperty(type)) {
+      p = {...acc[type]}
+      if (id && (type === 'buckets' || type === 'telegrafs')) {
+        if (p.sublevelPermissions.hasOwnProperty(id)) {
+          p.sublevelPermissions[id].permissions[action] = true
+        } else {
+          p.sublevelPermissions[id] = {
+            id: id,
+            orgID: orgID,
+            name: name,
+            permissions: {
+              read: action === 'read',
+              write: action === 'write',
+            },
+          }
+        }
+      } else {
+        p[action] = true
+      }
+    } else {
+      if (id && (type === 'buckets' || type === 'telegrafs')) {
```

Comment:
should this be

```ts
} else if (id && (type === 'buckets'....
```
?

Reply:
there are a lot of if else statement here so i get the confusion. but this one should remain `else` because this else statement is there in case a specific resource is not already in the accumulator. in the case where we are iterating over a new resource type, then we are checking if that object has an id. if so, we are structuring the new object differently. if it doesn't have an id, (line180 takes care of that) then we are creating an object with only 2 properties. 

first if statement --> if (acc already has a (bucket) property) .... do this. else { if (current object has an id AND type is bucket or telegraf) {do this ....} else { do this}}